api/nakadi-event-bus-api.yaml

```diff
@@ -606,22 +488,388 @@ paths:
+
+  /subscriptions/{subscription_id}/events:
+    get:
+      tags:
+        - stream-api
+        - managed-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        Starts a new stream for reading events from this subscription. The data will be automatically rebalanced
+        between streams of one subscription. The minimal consumption unit is a partition, so it is possible to create as
+        many streams as the total number of partitions in event-types of this subscription. The rebalance currently
+        only operates with the number of partitions so the amount of data in event-types/partitions is not considered
+        during autorebalance.
+        The position of the consumption is managed by Nakadi. The client is required to commit the cursors he gets in
+        a stream. For details of how committing works see description of parameters `commit_mode`, `commit_semantics`
+        and the description of commit endpoint.
+      parameters:
+        - $ref: '#/parameters/SubscriptionId'
+        - name: commit_timeout
+          in: query
+          description: |
+            The amount of time allowed between commits. If no commit is performed within this amount of time
+            - the connection will be closed from Nakadi side. Specified in seconds.
+          type: integer
+          format: int32
+          required: false
+          default: 30
+        - name: window_size
+          in: query
+          description: |
+            The amount of uncommitted events Nakadi will stream before pausing the stream. When in paused
+            state and commit comes - the stream will continue. Minimal window size is 1.
+          type: integer
+          format: int32
+          required: false
+          default: 10000
+        - name: commit_mode
+          in: query
+          description: |
+            The commit mode that will be used for this stream. Currently only `commit_at_will` is supported.
+            Possible values:
+
+            - `commit_at_will`: client is free to choose how often he wants to commit. Of course the commit_timeout
+            property should still be respected.
+            When a batch is committed that also automatically commits all previous batches that were sent in a stream
+            (if the commit_semantics parameter is `per_partition` - that will commit all previous batches only for
+            current partition).
+
+            - `commit_each`: this commit mode requires each batch to be committed, and it should be done in correct
+            order. Committing a cursor when the previous is not yet committed is not possible.
+
+            - `auto_commit`: everything that Nakadi sends to stream is considered to be committed. No actions are
+            required from client side to commit. Be aware that with this commit mode loosing of some events is very
+            probable. So you should know what you are doing if you use autocommit.
```

Comment:
hmmmm, i'm not sure that i would even offer this option -- might be too tempting for lazy developers ;)